### PR TITLE
Cleaned up duplicate dynamic type functions

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Consolidated both dynamic-type functions into a single one, separating logic based on `gMain.inBattle`.

Note: because `gBattleStruct->ateBoost` was being set during the "in battle" version of the function, this consolidated version has an `ateBoost` argument that accepts `NULL` to tell it to not do so in the summary in battles and such.

## Images
Tested in-game that dynamic move types still worked in UIs.
image image
image image

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
labels

File: src/battle_main.c [hunk 1]
```diff
@@ -5774,151 +5776,246 @@ bool32 TrySetAteType(u32 move, u32 battlerAtk, u32 attackerAbility)
...
             else if (gBattleWeather & B_WEATHER_SUN && holdEffect != HOLD_EFFECT_UTILITY_UMBRELLA)
-                gBattleStruct->dynamicMoveType = TYPE_FIRE | F_DYNAMIC_TYPE_SET;
-            else if (gBattleWeather & (B_WEATHER_HAIL | B_WEATHER_SNOW))
-                gBattleStruct->dynamicMoveType = TYPE_ICE | F_DYNAMIC_TYPE_SET;
+                return TYPE_FIRE;
+            else if (gBattleWeather & (B_WEATHER_SNOW | B_WEATHER_HAIL))
+                return TYPE_ICE;
             else
-                gBattleStruct->dynamicMoveType = TYPE_NORMAL | F_DYNAMIC_TYPE_SET;
+                return moveType;
```
Comment (AlexOn1ine) [inline src/battle_main.c:5824]: for consistency sake and readability could you turn this one into a switch like?

File: src/battle_main.c [hunk 2]
```diff
@@ -5774,151 +5776,246 @@ bool32 TrySetAteType(u32 move, u32 battlerAtk, u32 attackerAbility)
...
+            return TYPE_MYSTERY;
     }
-    else if (gMovesInfo[move].effect == EFFECT_RAGING_BULL
-            && (gBattleMons[battlerAtk].species == SPECIES_TAUROS_PALDEAN_COMBAT_BREED
-             || gBattleMons[battlerAtk].species == SPECIES_TAUROS_PALDEAN_BLAZE_BREED
-             || gBattleMons[battlerAtk].species == SPECIES_TAUROS_PALDEAN_AQUA_BREED))
+    else if (moveEffect == EFFECT_RAGING_BULL
+            && (species == SPECIES_TAUROS_PALDEAN_COMBAT_BREED
+             || species == SPECIES_TAUROS_PALDEAN_BLAZE_BREED
+             || species == SPECIES_TAUROS_PALDEAN_AQUA_BREED))
```
Comment (AlexOn1ine) [inline src/battle_main.c:5899]:
```suggestion
             && (species == SPECIES_TAUROS_PALDEAN_COMBAT_BREED
              || species == SPECIES_TAUROS_PALDEAN_BLAZE_BREED
              || species == SPECIES_TAUROS_PALDEAN_AQUA_BREED))
```

**Review by AlexOn1ine — COMMENTED** [review]
Looks very clean, just a few nitpicks

File: src/battle_main.c (hunk 1 above)
Comment (AsparagusEduardo) [reply src/battle_main.c:5824]: Can you do switches with bit operations?

Comment (AlexOn1ine) [reply src/battle_main.c:5824]: oh right, sorry

it is fine then

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**